File: Assets/CsProtocol/Websocket/WebsocketHelloResponse.cs
using System;
using System.Collections.Generic;
using CsProtocol.Buffer;

namespace CsProtocol
{
    
    public class WebsocketHelloResponse : IProtocol
    {
        public string message;

        public static WebsocketHelloResponse ValueOf(string message)
        {
            var packet = new WebsocketHelloResponse();
            packet.message = message;
            return packet;
        }


        public short ProtocolId()
        {
            return 1401;
        }
    }


    public class WebsocketHelloResponseRegistration : IProtocolRegistration
    {
        public short ProtocolId()
        {
            return 1401;
        }

        public void Write(ByteBuffer buffer, IProtocol packet)
        {
            if (buffer.WritePacketFlag(packet))
            {
                return;
            }
            WebsocketHelloResponse message = (WebsocketHelloResponse) packet;
            buffer.WriteString(message.message);
        }

        public IProtocol Read(ByteBuffer buffer)
        {
            if (!buffer.ReadBool())
            {
                return null;
            }
            WebsocketHelloResponse packet = new WebsocketHelloResponse();
            string result0 = buffer.ReadString();
            packet.message = result0;
            return packet;
        }
    }
}

File: Assets/zfoolua/websocket/WebsocketHelloResponse.lua

local WebsocketHelloResponse = {}

function WebsocketHelloResponse:new()
    local obj = {
        message = "" -- string
    }
    setmetatable(obj, self)
    self.__index = self
    return obj
end

function WebsocketHelloResponse:protocolId()
    return 1401
end

function WebsocketHelloResponse:write(buffer, packet)
    if packet == nil then
        buffer:writeInt(0)
        return
    end
    buffer:writeInt(-1)
    buffer:writeString(packet.message)
end

function WebsocketHelloResponse:read(buffer)
    local length = buffer:readInt()
    if length == 0 then
        return nil
    end
    local beforeReadIndex = buffer:getReadOffset()
    local packet = WebsocketHelloResponse:new()
    local result0 = buffer:readString()
    packet.message = result0
    if length > 0 then
        buffer:setReadOffset(beforeReadIndex + length)
    end
    return packet
end

return WebsocketHelloResponse
